spec/models/manageiq/providers/hawkular/middleware_manager/event_catcher/runner_spec.rb

```diff
@@ -80,4 +80,58 @@
       expect(hash[:full_data]).to be_an String
     end
   end
+
+  context "event_monitor" do
+    it "reset_event_monitor_handle" do
+      expect(subject.instance_variable_get('@event_monitor_handle')).to be_nil
+      subject.send(:reset_event_monitor_handle)
+    end
+
+    it "stop_event_monitor" do
+      expect(subject).to receive(:reset_event_monitor_handle)
+      expect(subject.instance_variable_get('@event_monitor_handle')).to be_nil
+      subject.send(:stop_event_monitor)
+    end
+
+    context "event_monitor_handle" do
+      it "event_monitor_handle not defined" do
+        VCR.use_cassette(described_class.name.underscore.to_s,
+                         :decode_compressed_response => true) do # , :record => :new_episodes) do
+          subject.instance_variable_set('@event_monitor_handle', nil)
+          subject.send(:event_monitor_handle)
+          expect(subject.instance_variable_get('@event_monitor_handle')).to be_an ManageIQ::Providers::Hawkular::MiddlewareManager::EventCatcher::Stream
+        end
+      end
+    end
+  end
+
+  it 'process_event' do
+    subject.instance_variable_set('@filtered_events', ['hawkular_event.critical', 'hawkular_datasource.ok'])
```

Comment:
@tumido I'm not sure but perhaps this would be influenced by the changes you are doing related to alerts/events?

Reply:
@abonas, that's just a spec, right? We might require to add specs for other events as well...